cpp/src/io/parquet/page_enc.cu

```diff
@@ -366,8 +369,9 @@ __global__ void __launch_bounds__(128)
             }
             page_g.max_hdr_size += stats_hdr_len;
           }
-          page_g.page_data        = ck_g.uncompressed_bfr + page_offset;
-          page_g.compressed_data  = ck_g.compressed_bfr + comp_page_offset;
+          page_g.page_data = ck_g.uncompressed_bfr + page_offset;
+          if (not comp_page_sizes.empty())
+            page_g.compressed_data = ck_g.compressed_bfr + comp_page_offset;
```

Comment:
Is it ok that `page_g.compressed_data` is only set if the page sizes isn't empty? This seems ok logically, but does something else expect it to be set to 0?

Reply:
This is a multi-pass kernel. In the pass where `comp_page_sizes` is empty, `pages` is also empty. So this has no effect.